src/App.tsx

```diff
@@ -62,11 +62,11 @@ function App() {
             onCancel={handleCancelButtonClick}
           />
         )}
-        {cardsArray.map((card, index) => (
+        {[...cardsArray].reverse().map(card => (
           <Card
             faceValue={card.face}
             flipValue={card.back}
-            key={index}
+            key={card.id}
```

Comment:
This change actually fixed a bug I had, which caused a situation where after deleting a card, the next card suddenly "inherited" the height of the deleted card. I guess this is a good example that shows how react renders and distinguishes between elements, and why keys should be unique not only to their "peers".